main.cpp

```diff
@@ -38,17 +41,16 @@ void find_k_to_end(struct Node* head, int k, int &result_value) {
         length++;
     }
     
-    if (length < k) {
-        return;
+    if (length < k - 1) {
+        return NULL;
     }
     
     curr = head;
-    for (i = 1; i < length - k + 1; ++i) {
+    for (i = 0; i < length - k - 1; ++i) {
```

Comment:
This will break if k < 0. Either check for it or stop this loop if you are about to follow a NULL pointer.